apis/docdb/v1alpha1/zz_db_cluster.go

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// DBClusterParameters defines the desired state of DBCluster
+type DBClusterParameters struct {
+	// Region is which region the DBCluster will be created.
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+	// A list of Amazon EC2 Availability Zones that instances in the cluster can
+	// be created in.
+	AvailabilityZones []*string `json:"availabilityZones,omitempty"`
+	// The number of days for which automated backups are retained. You must specify
+	// a minimum value of 1.
+	//
+	// Default: 1
+	//
+	// Constraints:
+	//
+	//    * Must be a value from 1 to 35.
+	BackupRetentionPeriod *int64 `json:"backupRetentionPeriod,omitempty"`
+	// The cluster identifier. This parameter is stored as a lowercase string.
+	//
+	// Constraints:
+	//
+	//    * Must contain from 1 to 63 letters, numbers, or hyphens.
+	//
+	//    * The first character must be a letter.
+	//
+	//    * Cannot end with a hyphen or contain two consecutive hyphens.
+	//
+	// Example: my-cluster
+	// +kubebuilder:validation:Required
+	DBClusterIdentifier *string `json:"dbClusterIdentifier"`
```

Comment:
We are able to use external-name annotation of the resource, so we need to make code generator [ignore](https://github.com/crossplane/provider-aws/blob/master/CODE_GENERATION.md#external-name) this field and we need to rely on that annotation in the code, which seems to be the case already except `postCreate`